### ISSUE TITLE
Api call to display completeness of initial sync

### ISSUE THREAD

**Issue by densone** [paradigmxyz/reth#4282]
### Describe the feature

In execution clients like erigon I can use eth_blockNumber to figure out where I am in the initial load up of the blockchain.  This does not seem to work for me in reth.  It just returns a 0 even though I can see the size of the data is growing. 

It would be nice to have some type of API call that can give us an idea of the block height. This allows companies like GitHub.com/blockjoy to be able to capture the current block in the sync. Here is an example. 

<img width="1508" alt="image" src="https://github.com/paradigmxyz/reth/assets/49484/28496dab-06de-43ec-a53e-8d61da801989">

and reth

<img width="1469" alt="image" src="https://github.com/paradigmxyz/reth/assets/49484/bd6fa6fa-bfb8-4b8a-859e-cb0d752dac20">

**Comment by gakonst** [comment]
@mattsse I know we've talked about this before...we probably need something like

```
reth_syncing
{
  "stages": "5/13"
  "stage": "execution", // "String" type
  "progress": {
        "gas_processed": "100"
        "total_gas": "1000"
   },
   "progress_pct": "10"
}
```

**Comment by gakonst** [comment]
Yes! Do you want to propose an API design based on the one I wrote above, and @mattsse and I can iterate with you on it before implementing?

**Comment by pyropy** [comment]
Hey @gakonst sure thing. I have also talked with Dragan about taking over #2836 so I don't know which would be bigger priority and maybe easier onboarding to reth codebase for first time contributor to reth codebase?

**Comment by gakonst** [comment]
Either works! Both are simple enough, so feel free to split the work as you wish. Happy to guide on directions for each ticket

**Comment by pyropy** [comment]
@gakonst I've already some things on #2836. Should I open new PR since #2897 is closed due to being stale?

**Comment by gakonst** [comment]
yup

**Comment by pyropy** [comment]
> yup

I'll pick it up on Monday as I am at Protocol Berg ATM

**Comment by gakonst** [comment]
Sick - please let us know if you need any help.

**Comment by jenpaff** [comment]
This should be solved by `eth_syncing`, feel free to open a new github issue if you still find a limitation
